### DISCUSSION TITLE
If variant is missing from an API download request, ORIGINAL should be selected as default

### DISCUSSION THREAD

**Discussion by jessienab** [LycheeOrg/Lychee#2650] [Ideas]
### Enhancement

At present, if a user attempts to download an image directly using it's ID, but without a variant chosen, an error page will be given.

Example:

https://photos . nabein . me/api/v2/Zip?photo_ids=q5MYBmAuLNJ61S5X_Qgqs9Ae

With no `?variant= `at the end, an error page is given. Instead, it might be better to auto-select the `ORIGINAL` variant, to avoid an error page appearing.

This would absolutely be an edge case, as I highly doubt many if any users would be downloading images directly based on IDs, but it was just a thought.

My only other thought relating to this, is that it could be used maliciously, for example people iterating over Base64 values to find valid photos, and since ORIGINAL would be invoked as the variant, it would drive up bandwidth usage (and thus costs), as well as CPU and other resource usage.

**Comment by ildyria** [comment]
Sure we can give it a default value, that is a very easy fix.
Probably around here:
https://github.com/LycheeOrg/Lychee/blob/master/app/Http/Requests/Album/ZipRequest.php#L63
and here:
https://github.com/LycheeOrg/Lychee/blob/master/app/Http/Requests/Album/ZipRequest.php#L77

**Comment by ildyria** [comment]
> This would absolutely be an edge case, as I highly doubt many if any users would be downloading images directly based on IDs, but it was just a thought.

I would not worry about that. There are more than 128 bits of randomness in those URLS.
It is the same complexity as trying to find the pre-images of a set of hashed random passwords.
If your number of picture is 2^n, your chance of success is 2^(128 - n)... 
With 64 thousands pictures, you have n = 16, so that does leave you with a nice margin.
